source/fuzz/fuzzer_pass_add_loads.cpp

```diff
@@ -91,35 +91,40 @@ void FuzzerPassAddLoads::Apply() {
         uint32_t memory_scope_id = 0;
         uint32_t memory_semtantics_id = 0;
 
-        // has work group or storage buffer (storage class)
-        if (chosen_instruction->GetInOperand(0).words[0] ==
-                SpvMemorySemanticsUniformMemoryMask ||
-            chosen_instruction->GetInOperand(0).words[0] ==
-                SpvMemorySemanticsWorkgroupMemoryMask) {
-          if (GetFuzzerContext()->ChoosePercentage(
-                  GetFuzzerContext()->GetChanceOfAddingAtomicLoad())) {
-            is_atomic_load = true;
-
-            memory_scope_id = FindOrCreateConstant(
-                {SpvScopeInvocation},
-                FindOrCreateIntegerType(32, GetFuzzerContext()->ChooseEven()),
-                false);
-
-            memory_semtantics_id = FindOrCreateConstant(
-                {static_cast<uint32_t>(
-                    chosen_instruction->GetInOperand(0).words[0] ==
-                            SpvMemorySemanticsUniformMemoryMask
-                        ? SpvMemorySemanticsUniformMemoryMask
-                        : SpvMemorySemanticsUniformMemoryMask)},
-                FindOrCreateIntegerType(32, GetFuzzerContext()->ChooseEven()),
-                false);
+        if (chosen_instruction->opcode() == SpvOpAccessChain) {
+          auto storage_class = GetIRContext()
+                                   ->get_def_use_mgr()
+                                   ->GetDef(chosen_instruction->type_id())
+                                   ->GetInOperand(0)
+                                   .words[0];
+          if (storage_class == SpvStorageClassUniform ||
+              storage_class == SpvStorageClassStorageBuffer ||
+              storage_class == SpvStorageClassWorkgroup ||
+              storage_class == SpvStorageClassFunction ||
+              storage_class == SpvStorageClassPrivate) {
```

Comment:
I don't think we want to insert atomic loads from _Function_ nor _Private_ storage classes. Check the SPIR-V specification to see which memory classes can be used with the memory semantics _UniformMemory_ and _WorkgroupMemory_.

Reply:
Removed. 
I thought need to add them for the `opLoad`.